### PR TITLE
enhancement/15026-new-approximation-demo

### PR DESCRIPTION
**Pull request by karolkolodziej**
Feature #15026, added simple approximation demo to the documentation.

### CODE REVIEW

File: samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        data: ADBE
+    }, {
+        name: 'ADBE start',
+        data: ADBE,
+        dataGrouping: {
+            approximation: function () {
+                var groupStart = this.dataGroupInfo.start;
+
+                return this.options.data[groupStart][1];
+            }
```
Comment (pawelfus) [inline samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.js:29]: Can be simpler :) `approximation` takes an argument (or many if data is e.g. OHLC), which is an array of values:
```js
            approximation: function (data) {
                return data[0];
            },
```

**Review by pawelfus — CHANGES_REQUESTED** [review]
See comment

File: samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                return groupData[0];
+            }
+        }
+    }, {
+        name: 'ADBE end',
+        data: ADBE,
+        dataGrouping: {
+            approximation: function (groupData) {
+                var groupEnd = this.dataGroupInfo.start +
+                     this.dataGroupInfo.length - 1;
```
Comment (pawelfus) [inline samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.js:35]: This doesn't work. `groupData` contains a list of values that should be used to calculate new, grouped value. For example: `[1, 2, 3]`, which means, DataGrouping wants to group these three points into one. Average would be 2, start=1, end=3.

`this.dataGroupInfo` is used when referring original dataset, not `groupData`.

Simple `return groupData[groupData.length - 1];` should be enough :)

**Review by pawelfus — CHANGES_REQUESTED** [review]

File: samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<div id="container" style="height: 400px; min-width: 600px"></div>
+
+<script src="https://code.highcharts.com/stock/highstock.js"></script>
+<script src="https://code.highcharts.com/stock/modules/exporting.js"></script>
+<script type="text/javascript" src="https://www.highcharts.com/samples/data/three-series-1000-points.js"></script>
```
Comment (raf18seb) [inline samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.html:5]: Is it necessary? `type="text/javascript"`

**Review by raf18seb — COMMENTED** [review]
I have only one small question, see the comment above.

File: samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.html (hunk 3 above)
Comment (karolkolodziej) [reply samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.html:5]: I think it's not necessary, I copied that from [approximation demo](https://jsfiddle.net/gh/get/library/pure/highcharts/highcharts/tree/master/samples/stock/plotoptions/series-datagrouping-approximation)

Comment (TorsteinHonsi) [reply samples/stock/plotoptions/series-datagrouping-simple-approximation/demo.html:5]: Not necessary.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Please fix the comment and merge in the master, then after CI, it should be ready to merge.

**Review by pawelfus — APPROVED** [review]
Nice! :)

**Review by pawellysy — APPROVED** [review]
Looks good!

**Review by raf18seb — APPROVED** [review]
All good, thanks!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**